vhdbuilder/packer/test/run-test.sh

```diff
@@ -29,24 +29,38 @@ trap cleanup EXIT
 DISK_NAME="${TEST_RESOURCE_PREFIX}-disk"
 VM_NAME="${TEST_RESOURCE_PREFIX}-vm"
 
-if [ "$MODE" == "sigMode" ]; then
-  echo "SIG existence checking for $MODE"
-  id=$(az sig show --resource-group ${AZURE_RESOURCE_GROUP_NAME} --gallery-name ${SIG_GALLERY_NAME}) || id=""
-  if [ -z "$id" ]; then
-    echo "Shared Image gallery ${SIG_GALLERY_NAME} does not exist in the resource group ${AZURE_RESOURCE_GROUP_NAME} location ${AZURE_LOCATION}"
-    exit 1
-  fi
+if [ "$MODE" == "default" ]; then
+  az disk create --resource-group $RESOURCE_GROUP_NAME \
+    --name $DISK_NAME \
+    --source "${OS_DISK_URI}" \
+    --query id
+  az vm create --name $VM_NAME \
+    --resource-group $RESOURCE_GROUP_NAME \
+    --attach-os-disk $DISK_NAME \
+    --os-type $OS_TYPE \
+    --public-ip-address ""
+else 
+  if [ "$MODE" == "sigMode" ]; then
+    id=$(az sig show --resource-group ${AZURE_RESOURCE_GROUP_NAME} --gallery-name ${SIG_GALLERY_NAME}) || id=""
+    if [ -z "$id" ]; then
+      echo "Shared Image gallery ${SIG_GALLERY_NAME} does not exist in the resource group ${AZURE_RESOURCE_GROUP_NAME} location ${AZURE_LOCATION}"
+      exit 1
+    fi
 
-  id=$(az sig image-definition show \
-    --resource-group ${AZURE_RESOURCE_GROUP_NAME} \
-    --gallery-name ${SIG_GALLERY_NAME} \
-    --gallery-image-definition ${SIG_IMAGE_NAME}) || id=""
-  if [ -z "$id" ]; then
-    echo "Image definition ${SIG_IMAGE_NAME} does not exist in gallery ${SIG_GALLERY_NAME} resource group ${AZURE_RESOURCE_GROUP_NAME}"
-    exit 1
-  fi
+    id=$(az sig image-definition show \
+      --resource-group ${AZURE_RESOURCE_GROUP_NAME} \
+      --gallery-name ${SIG_GALLERY_NAME} \
+      --gallery-image-definition ${SIG_IMAGE_NAME}) || id=""
+    if [ -z "$id" ]; then
+      echo "Image definition ${SIG_IMAGE_NAME} does not exist in gallery ${SIG_GALLERY_NAME} resource group ${AZURE_RESOURCE_GROUP_NAME}"
+      exit 1
+    fi
 
-  IMG_DEF="/subscriptions/${SUBSCRIPTION_ID}/resourceGroups/${AZURE_RESOURCE_GROUP_NAME}/providers/Microsoft.Compute/galleries/${SIG_GALLERY_NAME}/images/${SIG_IMAGE_NAME}/versions/${SIG_IMAGE_VERSION}"
+    IMG_DEF="/subscriptions/${SUBSCRIPTION_ID}/resourceGroups/${AZURE_RESOURCE_GROUP_NAME}/providers/Microsoft.Compute/galleries/${SIG_GALLERY_NAME}/images/${SIG_IMAGE_NAME}/versions/${SIG_IMAGE_VERSION}"
+  else 
```

Comment:
Could we use the check `if [ "$MODE" == "gen2Mode" ];` otherwise this mode will be regarded as the default matched mode and Windows won't get here.

Reply:
there are 3 build modes, 1) default 2) sigMode 3) gen2Mode

I dont see how Windows misses any required mode? Am I missing something?

Reply:
For now, Windows use only default and sigmode. But this is not a blocker.